### PR TITLE
Attack chains - support in in-cluster "security" framework scanning

### PR DESCRIPTION
**Pull request by kooomix**
## Overview
Support in in-cluster scanning of "security" framework for attack-chain.
If env var KS_SCAN_SECURITY_FRAMEWORK in the in-cluster kubescape pod is True then "security" framework will be scanned on top of any framework scan (requires updating helm in order to activate).

### Updating regolibrary (GitRegoStore) to version **v1.0.286-rc.0**:
- version includes a new file _security_frameworks_.
-  _security_frameworks_ is expected to exist only from version **v1.0.286-rc.0** and above for backward compatibility.

### New framework attribute "typeTags":
- a list of framework types, currently can be "compliance" and/or "security".
- for compliance framework, typeTags is ["compliance"]
- for security framework, typeTags is ["security"]
- if no typeTags is configured or is empty, it will be considered as "compliance"

## How to Test
- CLI scan should never scan security framework unless specifically asked for this framework (`kubescape scan framework security`)
- Once helm is updated with the env var as True, any in-cluster framework scan will include also the security framework.

## Checklist before requesting a review

put an [x] in the box to get it checked 

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: httphandler/handlerequests/v1/datastructuremethods.go
Comment (dwertent) [inline httphandler/handlerequests/v1/datastructuremethods.go:94]: What is the use-case?

File: core/cautils/getter/testdata/MITRE.json
Comment (dwertent) [inline core/cautils/getter/testdata/MITRE.json:9]:
```suggestion
  "categories": ["compliance"],
```
I think this is a more appropriate name.

**Review by dwertent — CHANGES_REQUESTED** [review]
Is this ready to merge? Are you sure there are no other attack-chain related PRs?

File: httphandler/handlerequests/v1/datastructuremethods.go
Comment (dwertent) [reply httphandler/handlerequests/v1/datastructuremethods.go:94]: I will explain: As we spoke, kubescape receives the list of frameworks from the Operator. So why does KS need to handle this at all?

File: core/cautils/getter/testdata/MITRE.json
Comment (kooomix) [reply core/cautils/getter/testdata/MITRE.json:9]: @dwertent 
We have used the same existing terminology as used for controls: see [this](https://github.com/search?q=repo%3Akubescape%2Fopa-utils%20ControlTypeTagCompliance&type=code).

File: httphandler/handlerequests/v1/datastructuremethods.go
Comment (kooomix) [reply httphandler/handlerequests/v1/datastructuremethods.go:94]: @dwertent 
The requirement is to **always** scan the security framework when scanning framework(s). This function is the main junction all in-cluster scans go through so it is not only easier and faster to implement, but also making sure all scans requested will have the security framework appended.

**Comment by kooomix** [comment]
> Is this ready to merge? Are you sure there are no other attack-chain related PRs?

There is a dependency on [this](https://github.com/armosec/cadashboardbe/pull/389) PR which effects only the case a user explicitly scanned "secuitry" fw - then it will appear on UI. The backend PR is meant to show only compliance scans on UI.

To save time, I planned to have this ks PR approved and merge it once the backend PR is successfully merged to all envs.

**Comment by kooomix** [comment]
> > Is this ready to merge? Are you sure there are no other attack-chain related PRs?
> 
> There is a dependency on [this](https://github.com/armosec/cadashboardbe/pull/389) PR which effects only the case a user explicitly scanned "secuitry" fw - then it will appear on UI. The backend PR is meant to show only compliance scans on UI.
> 
> To save time, I planned to have this ks PR approved and merge it once the backend PR is successfully merged to all envs.

Correction - already now users can scan "security" fw (as it was added to the regolibrary release) so there is no dependency on the PR I mentioned.

File: httphandler/handlerequests/v1/datastructuremethods.go
Comment (dwertent) [reply httphandler/handlerequests/v1/datastructuremethods.go:94]: Correct, but the requirements do not specify how to implement...
By definition, Kubescape is a worker, meaning, kubescape carries out what was asked of him todo, without making "decisions" of what to run and how. For this reason, I expect the Operator/BE who triggers the scanning to handle such logic.

File: core/cautils/getter/testdata/MITRE.json
Comment (dwertent) [reply core/cautils/getter/testdata/MITRE.json:9]: 1. The JSON is user-facing (unlike the code)
2. I will also change the variables naming, but it is not dependant.

Comment (kooomix) [reply core/cautils/getter/testdata/MITRE.json:9]: @dwertent 
1. Not sure what you mean here - if you mean that the user will eventually see the typeTags field, i'm ok with that.

File: httphandler/handlerequests/v1/datastructuremethods.go
Comment (Bezbran) [reply httphandler/handlerequests/v1/datastructuremethods.go:94]: This is a requirement for the worker. Not for the orchestrator.

**Pull request closed without merging**
